### PR TITLE
Feature/cypress

### PR DESCRIPTION
**Pull request by hvohr**
### Summary:
- Describe what you did
- Added cypress testing for the project, including sad path testing and error handling. I also fixed an issue where a previous dependency being used was outdated, so I refactored and added a new npm package to replace and repair this issue. Mock server endpoints are being used for the tests and will eventually need to be replaced. I also added commented out tests for when we implement post.
- Any instructions for the reviewer?
- Run `npm install` and when opening cypress, use `npx cypress open`
### Did this break anything?
- [x] No
- [ ]  Yes
### Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x]  Bug fix (non-breaking change which fixes an issue)
- [x]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
### Checklist:
- [x]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [x]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

### CODE REVIEW

File: app/Components/CreateProblemModal.js [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+  //   { label: "V7", value: "V7" },
+  //   { label: "V8", value: "V8" },
+  //   { label: "V9", value: "V9" },
+  //   { label: "V10", value: "V10" },
+  //   { label: "V11", value: "V11" },
+  //   { label: "V12", value: "V12" },
+  //   { label: "V13", value: "V13" },
+  //   { label: "V14", value: "V14" },
+  //   { label: "V15", value: "V15" },
+  // ];
```
Comment (Jade-ZS) [inline app/Components/CreateProblemModal.js:31]: MP: seems that we got some comments left out here, but we can definitely fix this in the next branch too!

**Review by Jade-ZS — APPROVED** [review]
Thanks for working on this Hollis! Testing is a lot of work and you did great! I like our new difficulty select menu and I'm glad it worked out! 

MP: There are some commented out code in CreateProblemModal and some of the test files, but we can definitely fix it in the another branch too!

File: app/Components/CreateProblemModal.js (hunk 1 above)
Comment (hvohr) [reply app/Components/CreateProblemModal.js:31]: Ooof totally left this, maybe whoever adds the post test can delete this too?? Or if I end up doing that I can delete it?

Comment (jalbe0076) [reply app/Components/CreateProblemModal.js:31]: I can take care of that

**Pull request merged**